tvm/src/target/source/codegen_opencl.cc

```diff
@@ -408,13 +409,31 @@ void CodeGenOpenCL::VisitExpr_(const CallNode* op, std::ostream& os) {
     } else {
       os << ss.str();
     }
+    lowering_texture2d_load_ = false;
   } else if (op->op.same_as(builtin_call_extern_)) {
     auto func = Downcast<StringImm>(op->args[0]);
     // Enable atomics extension if used.
     if (func->value == "atomic_add") {
       enable_atomics_ = true;
     }
     CodeGenC::VisitExpr_(op, os);
+  } else if (op->op.same_as(builtin::if_then_else())) {
```

Comment:
I think that using `SelectNode` visitor is more proper way to do such changes. Please take a look on this PR: https://github.com/apache/tvm/pull/11038/files
Also, on the `tvm/main` we use `select` for all data types. Probably, we could remove `lowering_texture2d_load_` variable.

Reply:
Thanks for the advice - that would be cleaner. In that case, do we need to transform `if_then_else` to `SelectNode`? In this repo, `if_then_else` seems preserved until the codegen phase - that is why I am directly handling `if_then_else`. I haven't checked `tvm/main` yet, does it transform `if_then_else` to `SelectNode`?

Reply:
Thanks for the advice - that would be cleaner. In that case, do we need to transform `if_then_else` to `SelectNode`? In this repo, `if_then_else` seems preserved until the codegen phase - that is why I am directly handling `if_then_else`. I haven't checked `tvm/main` yet, does it transform `if_then_else` to `SelectNode`?

Reply:
@lhez, probably I wasn't right and overriding `SelectNode` won't be enough for your task. Please, check it.